crypto/evp/kdf_meth.c

```diff
@@ -136,10 +142,13 @@ static void *evp_kdf_from_algorithm(int name_id,
             if (kdf->set_ctx_params != NULL)
                 break;
             kdf->set_ctx_params = OSSL_FUNC_kdf_set_ctx_params(fns);
+            if (kdf->set_ctx_params == NULL)
+                break;
+            fnctxcnt++;
```

Comment:
If there is a set_ctx_params, settable_ctx_params is mandatory. However is actually set_ctx_params mandatory? I think there could in theory be a KDF which needs only a key but has otherwise no params.

Reply:
That could well be, but the KDF docs say a set_ctx_params method is required.  If there there theoretically could be a kdf provider that doesn't require any params, I would imagine we should remove the requirement from the docs and do a null check before calling the method pointer

Reply:
Is there a consensus here?  should we remove the requirement for a set_ctx_params method from the documentation?

Reply:
Hm... I'd rather not start enforcing something that was not enforced before and could have worked. BTW, params could be set also through the init call so IMO set_ctx_params should definitely not be mandatory -> doc fix.

Reply:
ok, I'll alter this PR to just do the doc fix, and create a new one to fixup all the required param checks

Reply:
Given the above audit, I don't think a separate PR is needed, I can just update this one to include checking in the currently unchecked providers and document them

Reply:
I still believe the cleanup NULL check is something very much different from checking the provided DISPATCH calls for being NULL.

Reply:
Instead of making set_ctx_params() mandatory, I would change the documentation to reflect the current reality.